src/setup.ts

```diff
@@ -33,13 +33,15 @@ async function setup(): Promise<void> {
 
         // download Notation CLI and validate checksum
         const downloadURL = getNotationDownloadURL(version, notation_url);
-        console.log(`Downloading Notation CLI from ${downloadURL}`);
+        console.log(`downloading Notation CLI from ${downloadURL}`);
         const pathToTarball: string = await tc.downloadTool(downloadURL);
-        if (notation_url) {
-            await validateCheckSum(pathToTarball, notation_checksum);
-        } else {
-            await validateCheckSum(pathToTarball, getNotationCheckSum(version)); 
+        console.log("downloading Notation CLI completed")
+        const sha256 = await hash(pathToTarball);
+        const groundTruth = notation_url ? notation_checksum : getNotationCheckSum(version);
```

Comment:
What's the error message if `version` is provided but not found?

Reply:
This scenario is covered in the e2e test as well. When the version is not found, the logic won't reach to this point. It will fail during `getNotationDownloadURL` at line 35 because such URL doesn't exist.
If `getNotationCheckSum` is used independently, then an unsupported version would still throw an error: `Notation CLI release does not support user input version ${version}`.